monitoring/monitoring.go

```diff
@@ -92,46 +88,60 @@ type Monitor struct {
 
 	sync.Mutex
 
-	currentSystemData pb.SystemMonitoring
+	globalMonitoringData cloudprotocol.GlobalMonitoringData
 
 	alertProcessors *list.List
 
-	serviceMap        map[string]*serviceMonitoring
-	trafficMonitoring TrafficMonitoring
+	instanceMonitoringMap map[string]*instanceMonitoring
+	trafficMonitoring     TrafficMonitoring
 }
 
-// ServiceMonitoringConfig contains info about service and rules for monitoring alerts
-type ServiceMonitoringConfig struct {
-	ServiceDir    string
-	IPAddress     string
-	UID           uint32
-	GID           uint32
-	UploadLimit   uint64
-	DownloadLimit uint64
-	ServiceRules  *ServiceAlertRules
+// MonitorParams instance monitoring parameters.
+type MonitorParams struct {
+	cloudprotocol.InstanceIdent
+	UID        uint32
+	GID        uint32
+	AlertRules *aostypes.ServiceAlertRules
 }
 
-type serviceMonitoring struct {
-	serviceDir             string
+type instanceMonitoring struct {
 	uid                    uint32
 	gid                    uint32
-	monitoringData         pb.ServiceMonitoring
+	monitoringData         cloudprotocol.InstanceMonitoringData
 	alertProcessorElements []*list.Element
 }
 
-/*******************************************************************************
+type processInterface interface {
+	Uids() ([]int32, error)
+	CPUPercent() (float64, error)
+	MemoryInfo() (*process.MemoryInfoStat, error)
+}
+
+/***********************************************************************************************************************
  * Variable
- ******************************************************************************/
+ **********************************************************************************************************************/
 
-// ErrDisabled indicates that monitoring is disable in the config
+// ErrDisabled indicates that monitoring is disable in the config.
 var ErrDisabled = errors.New("monitoring is disabled")
 
-/*******************************************************************************
+// These global variables are used to be able to mocking the functionality of getting quota in tests.
+// nolint:gochecknoglobals
+var (
+	systemCPUPersent    = cpu.Percent
+	systemVirtualMemory = mem.VirtualMemory
+	systemDiskUsage     = disk.Usage
+	getUserFSQuotaUsage = platform.GetUserFSQuotaUsage
+	processesInterface  []processInterface
```

Comment:
Mock get processes function instead of values it returns.

Reply:
Done. use function instead of array